portal-api/src/v1/helpers/date.ts

```diff
@@ -46,17 +46,38 @@ export const getStartOfDateFromDayMonthYearStrings = (day: string, month: string
  * This function has odd behaviour inherited from moment js:
  *  - If the month is invalid return NaN
  *  - If the day/year is invalid, use the current day/year instead
+ *  - If the current date is 29th February, and setting year to a non-leap year, use 28th instead
+ *  - If the current date is too large for the given month (e.g. 31st November), wrap (e.g to 1st December)
  */
 export const getStartOfDateFromDayMonthYearStringsReplicatingMoment = (day: string, month: string, year: string) => {
   // moment().set() returns invalid date if the month is invalid
   if (Number.isNaN(Number(month))) {
     return new Date(NaN);
   }
+
+  const currentYear = new Date().getFullYear();
+  const currentDate = new Date().getDate();
+  const parsedYear = Number.isNaN(Number(year)) ? currentYear : Number(year);
+
+  // If the current date is 29th February, and the parsedYear is not a leap year should use the current date as 28 instead
```

Comment:
add ticket ref to this comment to fix it

Reply:
comment about this being moment behaviour because i was far too silly to understand this

Reply:
Have moved comment to inside at top of method